src/app/3d/qgslightswidget.cpp

```diff
@@ -99,6 +152,23 @@ void QgsLightsWidget::onAddLight()
   mPointLights << QgsPointLightSettings();
   updateLightsList();
   cboLights->setCurrentIndex( cboLights->count() - 1 );
+  // To set default parameters of the light
+  onCurrentDirectionalLightChanged( 0 );
+}
+
+void QgsLightsWidget::onAddDirectionalLight()
+{
+  if ( mDirectionalLights.count() >= 4 )
```

Comment:
```suggestion
  if ( mDirectionalLights.count() > 4 )
```

If it should reflect the warning below.